docker/Dockerfile

```diff
@@ -1,14 +1,26 @@
+# In the first stage, create our non-privileged user and directories
 FROM alpine:3.16
 
 ENV SERVICE_HOME=/opt/palantir/services/spark-scheduler
-ENV SPARK_SCHEDULER_STARTUP_CONFIG=$SERVICE_HOME/var/conf/install.yml
 
 RUN adduser -S -u 5001 spark-scheduler && \
     mkdir -p $SERVICE_HOME && \
     mkdir -p $SERVICE_HOME/service/bin && \
     mkdir -p $SERVICE_HOME/var/conf && \
     chown -R 5001:5001 $SERVICE_HOME
 
+FROM scratch
+
+ENV SERVICE_HOME=/opt/palantir/services/spark-scheduler
+
+# Copy the spark-scheduler user from the alpine stage
+COPY --from=0 /etc/passwd /etc/passwd
+
+# Copy the service home from the alpine stage
+COPY --from=0 $SERVICE_HOME $SERVICE_HOME
```

Comment:
nit: can't we just `ENV SERVICE_HOME=/opt/palantir/services/spark-scheduler` in this stage only? Same goes for SSCW image

Reply:
Ah nvm, we have to copy that because only `alpine` knows how to `mkdir`.